mautic/mautic
# ownerfield tokens

closes #7964 
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR is follow up to https://github.com/mautic/mautic/pull/7066 by @isleshocky77 

Reworked with better performance

Tokens
- `{ownerfield=firstname}`
- `{ownerfield=lastname}`
- `{ownerfield=email}`
- `{ownerfield=position}`
- `{ownerfield=signature}`

This PR also fixed tokens in Send custom email from contact profile. 
Now If you have enabled Mailer is owner and contact has owner, then owner data will be tokenized

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create email with tokens above
3. Create contact without owner and with owner
4. Try Send email from contact profile
5. Try send segment email
6. Try send campaign email

Check If owner's details are filled properly as you expected.
